ModPatches/Rimsenal Federation/Defs/Rimsenal Federation/ThingDefs_Misc/Ammo_Fed.xml

```diff
@@ -1,9 +1,53 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Defs>
 
+	<!-- ==========  Bases =========== -->
+	
+	<ThingDef Name="BaseOpticBoltCE" ParentName="BaseBulletCE" Abstract="true">
+		<label>energy bolt</label>
+		<graphicData>
+			<texPath>Things/Projectile/Mlaser</texPath>
+			<graphicClass>Graphic_Single</graphicClass>
+		</graphicData>
+		<projectile Class="CombatExtended.ProjectilePropertiesCE">
+			<damageDef>Optic</damageDef>
+			<speed>100</speed>
+			<dropsCasings>false</dropsCasings>
+		</projectile>
+	</ThingDef>
+	
+	<ThingDef Name="BaseCrucibleBoltCE" ParentName="BaseBulletCE" Abstract="true">
+		<label>energy bolt</label>
+		<graphicData>
+			<texPath>Things/Projectile/Cruciblebolt</texPath>
+			<graphicClass>Graphic_Single</graphicClass>
+		</graphicData>
+		<projectile Class="CombatExtended.ProjectilePropertiesCE">
+			<damageDef>Crucible</damageDef>
+			<speed>100</speed>
+			<dropsCasings>false</dropsCasings>
+		</projectile>
+	</ThingDef>
+	
+	<ThingDef Name="BasePlasmaBombCE" ParentName="BaseExplosiveBullet" Abstract="true">
+		<label>plasma sphere</label>
+		<thingClass>CombatExtended.ProjectileCE_Explosive</thingClass>
+		<graphicData>
+			<texPath>Things/Projectile/Plasma</texPath>
+			<graphicClass>Graphic_Single</graphicClass>
+		</graphicData>
+		<projectile Class="CombatExtended.ProjectilePropertiesCE">
+			<suppressionFactor>6.0</suppressionFactor>
+			<dangerFactor>4.0</dangerFactor>
+			<damageDef>Plasma</damageDef>
+			<speed>16</speed>
```

Comment:
I think this is going to be too slow, given the reasonably long range of the weapons that use it like the plasma rifle--the target can comfortably be 4 - 5 tiles if it starts moving as you fire, and leading a target that's constantly making tiny course changes as it navigates rough terrain is a crap shoot.

Reply:
The plasma rifle itself is actually the longest range, with the other launchers generally being on the shorter end for their sort of weapon.
Additionally the reduced gravity factor the plasma projectiles have mitigates the effect of the speed somewhat and the flat trajectory still allows for incidental hits since they mostly fly at target height across their path, creating a zone of suppression thus limiting target movements.
Finally, the plasma guns are suppsoed to be high strength anti-static weapons, even in vanilla it is trivial to walk out of their path if you have the space to do so, the risk being that in order to do that you have to be out in the open to do so and as such succeptable to every non plasma armament in the raid (and even still, the plasma spheres providing suppresson means that if you screw up you'll get locked in place and killed).